### DISCUSSION TITLE
How to check if  js grid extension has been added?

### DISCUSSION THREAD

**Discussion by Vots1000** [PrestaShop/PrestaShop#35611] [Q&A]
Hello.
If You are using a component there is a lock not to init it again, and what about Grid.addExtension?
In the documentation there is only:

```
const grid = new window.prestashop.component.Grid(gridId);
grid.addExtension(new window.prestashop.component.GridExtensions.ColumnTogglingExtension());
```

It works, but if you add it several times actions will be duplicated.
Is there a way to check for extensions already added to grid to prevent this ?

**Comment by kpodemski** [comment]
That's a great question! I don't think there is any. That sounds like a good improvement to make!

**Reply by matks** [reply]
Agreed, this usecase was not anticipated

**Comment by Vots1000** [comment]
While waiting for the right answer, I was searching for a solution. 
So far, I have come up with this:
```
$(document).ready(function () {
    let firstClass = document.querySelector("#order_grid_table > tbody > tr:nth-child(1) > td.toggle-type");
    if (null != firstClass && firstClass.classList.contains('--my-class-name')) {
        const AsyncToggleColumnGridExt = new window.prestashop.component.Grid('order');
        AsyncToggleColumnGridExt.addExtension(new window.prestashop.component.GridExtensions.AsyncToggleColumnExtension());
    }
});
```
It's not perfect, but works by preventing to use addExtension() more then once.
Maybe it will be useful for introducing more universal solution.
